src/main/kotlin/com/workos/user_management/builders/CreateOrUpdateUserOptionsBuilder.kt

```diff
@@ -0,0 +1,84 @@
+package com.workos.user_management.builders
+
+import com.workos.user_management.types.CreateOrUpdateUserOptions
+import com.workos.user_management.types.PasswordHashTypeEnumType
+
+/**
+ * Builder for options when creating a user.
+ *
+ * @param email The email address of the user.
```

Comment:
`email` is only a possible option when creating a user. Should we have separate objects for create and update to make sure people don't attempt to pass `email` when updating a user?

Reply:
Might make sense to have some kind of `BaseCreateOrUPdateUserOptions` class that both can inherit from with only the common options, but I'll defer to you as the more experienced Kotlin developer.

Reply:
OMG good call... for some reason I read from the docs that both create and update were operating off of the email. I just made the change, and also pushed an `AbstractUserOptionsBuilder<OptionsObject>` to simplify the code of both create/update builders.